### PR TITLE
memory leak fixes

### PR DESCRIPTION
**Pull request by syamsidhardh**
* Fix few memory leak, fd leak, unutilized variables

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5e9ad303627f5e1d4c733077bc4dbafc33b0d7d7] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5e9ad303627f5e1d4c733077bc4dbafc33b0d7d7] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@syamsidhardh Thank you for contribution. But could you check belows?
1. Please write a commit description per a commit to know why what how (details).
2. Please write an exact location of module where you modify at commit title
3. Please write where and what you modify at commit title, not a result.
Those are written in our wiki.
https://github.com/Samsung/TizenRT/wiki/Documentations#commit-rules
```
wireless/scsc: Fix uninitialized variable
slsi_wifi: Fix double fd close
framework/st_things: free abs_json_path before exit
framework/wifi_manager: close opened dir before exit
```

File: framework/src/st_things/things_stack/things_stack.c [hunk 1]
```diff
@@ -252,6 +258,7 @@ int things_initialize_stack(const char *json_path, bool *easysetup_completed)
 
 	if (!dm_init_module(abs_json_path)) {
 		THINGS_LOG_E(TAG, "dm_init_module() failed");
+		things_free(abs_json_path);
```
Comment (Hyunjun85) [inline framework/src/st_things/things_stack/things_stack.c]: Could you move the common codes below '#endif'?

**Comment by syamsidhardh** [comment]
Uploaded a patchset which addresses the comments provided by @sunghan-chang .
Thanks.

**Comment by seinfra** [comment]
Target : [4090e0842d15383cee3a38e35c73ba3115dc25f4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4090e0842d15383cee3a38e35c73ba3115dc25f4] - Code Rule Check (C++) OK.

**Pull request merged**